astropy/coordinates/tests/test_frames.py

```diff
@@ -890,8 +890,8 @@ def test_equal():
     ne = sc1 != sc2
     assert np.all(eq == [True, False])
     assert np.all(ne == [False, True])
-    assert (sc1[0] == sc2[0]) == True  # noqa  (numpy True not Python True)
-    assert (sc1[0] != sc2[0]) == False  # noqa
+    assert sc1[0] == sc2[0]
+    assert not (sc1[0] != sc2[0])
```

Comment:
I would like coordinates maintainers to have a look at this one: @StuartLittlefair, @adrn, @eteq, @eerovaher, @mhvk (according to Team page)

Reply:
I think the point here was indeed to show that `__eq__` returns a boolean (and not something like `"boo!"`, which would evaluate like true...), so I'd leave this one as it was (adding the relevant warning code).

Reply:
If the intention is to test that the return values are indeed Booleans then the correct way to assert that would be
```python
assert (sc1[0] == sc2[0]) is True
assert (sc1[0] == sc2[0]) is False
```

Reply:
Yes, but note the comment in the original line: one gets a numpy True, not a python one...

Reply:
Would it make sense to write `assert (sc1[0] == sc2[0]) is np.bool_(True)`? Then the test is really insisting on getting the numpy True.

Reply:
We really should implement @taldcroft's suggestion because the current test does not distinguish between Python and `numpy` True:
```python
>>> import numpy as np
>>> np.bool_(True) == True
True
```

Reply:
The test is really just to show that `__eq__` returns something like a boolean (i.e., not `sc` or `"boo!"` that evaluates as true). I think it is an implementation detail whether that is a numpy or python bool, so would keep the test as it was originally. Or do `assert isinstance(sc1[0] == sc2[0], (bool, np.bool_))` -- that makes the intent clearer.

Reply:
See 757d1d9fd8c8b55ade21f6dc2d566c9ca5e8047f

Reply:
While reviewing the black PR for io.fits I see this line 
```
        assert isinstance(v := (tbhdu.columns.columns[4].array[0] == np.True_), (bool, np.bool_)) and v
```

which brings me here through git blame, to ask ... whyyy ?? 
The original code was just testing that an array element was a True value. Why does it need to be so complicated just to avoid a `noqa` flag ?

Reply:
@saimn, wen I originally "fixed" this, i attempted to change this to `assert tbhdu.columns.columns[4].array[0] is np.True_` (the `noqa` flag was suppressing the suggestion to change this). @mhvk then identified this as a test which was testing the result of the `__eq__` function; meaning that, I needed to change this back. Moreover, @mhvk noted that there are lots of return values from `__eq__` which will technically don't fail the `assert` such as returning `"boo!"`, which means we ought to test that the result is a boolean (the `bool` vs `np.bool_` type was an implementation detail). Hence the result turned into a test of the type of the equality return value.

After thinking about this, the original tests were essentially testing that the array elements were `True` (under the assumption that `__eq__` would be returning a boolean). Which means that the new `isinstance` check would not be able to detect if "True" and "False" were accidentally reversed in the `__eq__` evaluation (or some other pathological bug), which the previous iteration of the test could in fact detect  (assuming the result was a boolean). So I added the walrus assignment to test that the `__eq__` was also the expected outcome in addition to the right type. This was probably overkill, but it means the adjustments to the test should cover all the same cases as before in addition to confirming the boolean nature of the the result.

Feel free to change it back if you would like.